lancie-dialog.html

```diff
@@ -134,14 +141,17 @@
 
         /**
         * Closes the lancie-dialog
+        * If there is a dismissCallback defined, it will only dismiss the dialog of said callback returns `true`
         * @return {void}
         */
         close: function () {
           setTimeout(function () {
-          this.style.display = 'none';
-            this.$.actualDialog.style.display = 'none';
-            this.$.backdrop.style.display = 'none';
-            this.$.backdrop.style.opacity = '0.0';
+            if (this.dismissCallback && this.dismissCallback() || !this.dismissCallback) {
```

Comment:
Instead of a function, it is easier to use an event. For example:

```js
const event = new CustomEvent('should-dismiss-dialog', {composed: true, bubbles: true});
event.shouldDismiss = true;
this.dispatchEvent(event);

if (event.shouldDismiss) {
  <...>
}
```

Then if you want to prevent the dismiss this, do:

```js
this.addEventListener('should-dismiss-dialog', (e) => {
  e.shouldDismiss = false;
});